src/query/functions/src/scalars/arithmetic.rs

```diff
@@ -498,6 +498,35 @@ fn register_unary_minus(registry: &mut FunctionRegistry) {
                     |a, _| -(AsPrimitive::<T>::as_(a)),
                 );
             }
+            NumberClass::UInt64 => {
+                registry
+                    .register_passthrough_nullable_1_arg::<NumberType<u64>, NumberType<i64>, _, _>(
+                        "minus",
+                        |_, val| {
+                            let min = (val.max as i128).wrapping_neg();
+                            let max = (val.min as i128).wrapping_neg();
+
+                            if min < std::i64::MIN as i128 || max > std::i64::MAX as i128 {
+                                return FunctionDomain::MayThrow;
+                            }
+
+                            FunctionDomain::Domain(SimpleDomain::<i64> {
+                                min: min as i64,
+                                max: max as i64,
+                            })
+                        },
+                        vectorize_with_builder_1_arg::<NumberType<u64>, NumberType<i64>>(
+                            |a, output, ctx| {
+                                if a < std::i64::MIN as i128 || a > std::i64::MAX as i128 {
```

Comment:
```suggestion
                                if a < -(i64::MIN as i128) as u64 {
```
Eventually, llvm will fold it to a u64 constant.